tools/cli/commands/generate.js

```diff
@@ -189,11 +189,30 @@ export function getQuestions( type ) {
 	const packageQuestions = [];
 	const jsPackageQuestions = [];
 	const pluginQuestions = [
+		{
+			type: 'list',
+			name: 'versioningMethod',
+			message: 'How do you want versioning to work for your plugin?',
+			choices: [
+				{
+					name:
+						'WordPress-style: Like 1.2, with each non-bugfix release always incrementing by 0.1.',
```

Comment:
```suggestion
						'WordPress-style (recommended): Like 1.2, with each non-bugfix release always incrementing by 0.1.',
```

Reply:
Shrug. Personally I'd recommend semver instead for new projects. Even WordPress core seems to have had to explain that 5.9→6.0 wasn't any more important than 5.8→5.9 or 6.0→6.1 because people outside of the WordPress sphere expect that.

But since it could really go either way, depending on whether the developers are more WordPress-y or other-y, I left out any recommendation other than putting "WordPress-style" first in the list.

Reply:
They're building a WordPress plugin with this command so we should encourage WordPress conventions when possible.

Reply:
[Looks to me like WordPress explicitly _doesn't_ establish a convention for plugin versioning.](https://core.trac.wordpress.org/ticket/40663#comment:3)

Reply:
I understand. The root idea was to strive for consistency. If it helps to explain the perspective, change my last comment by s/WordPress/Jetpack and reflect on who actually uses this command to scaffold plugins and where those plugins live and are released from. The context of this patch came from a desire to simplify our release tooling and cognitive load for internal plugin developers. Does it help achieve that goal? It might help provide consistency with opinionated docs and tools. However this particular opinion is not a hill I care enough to die on, and it's a bit late for it anyway

Reply:
I don't think we'll be able to have consistency here, as we already have established plugins using both methods.

I'm not very concerned about potentially simplifying the tooling in this respect, in almost all cases the tooling just calls changelogger to handle it.

Cognitive load depends on how often people need to actually know what the next version number is going to be, and deal with more than just one of the plugins that they're already familiar with. I don't know the answer to that, if it's rare or common.

The problem I was trying to solve with this PR is that it seems people who've created plugins may have been unaware that the setting for WP-style versioning versus semver exists, resulting in it defaulting to semver where they want WP-style. Directly presenting the option is hopefully enough to resolve that for new plugins; for the existing plugins I'm hoping we'll directly ask in a P2 post or something.

Reply:
> I don't think we'll be able to have consistency here, as we already have established plugins using both methods.

Since we're also aiming for consistency among WordPress.org plugins, can we just say that in the description as a stalemate breaker?

"WordPress-style (recommended for plugins)"

Reply:
This question is only asked for plugins in the first place.

Reply:
Even better, let's commit the suggestion then and go with that.

Reply:
I'd still rather not. What's the basis for making a recommendation? Neither method is objectively better. We're not going to actually have consistency, tooling is already pretty simple in this respect (and without mandatory consistency it can't be simplified anyway), cognitive load would need research to determine the extent to which it's an issue, and WordPress core explicitly doesn't take a side on it. It comes down to which way the developers actually working on the plugin think about versioning for it.

The only real problem seems to be that some teams that do want WP-style don't seem to have known it's an option and have been trying to work around it.

Reply:
> The only real problem seems to be that some teams that do want WP-style don't seem to have known it's an option and have been trying to work around it.

Yeah, this is the big issue that we're trying to fix. I understand your point - if WordPress versioning isn't required maybe a team would actually prefer to use Semver, but in my (granted anecdotal) experience, most people are expecting WordPress style versioning because that's what Jetpack uses, and people are looking to Jetpack as an example. When the tooling uses semver, they get confused and then have to update things manually.

So maybe to reframe Derek's suggestion here - this standard isn't something WordPress requires, but it could be something Jetpack and Jetpack products require for consistency. But until that determination is made, we allow both.

As a compromise, maybe `recommended` is too strong, but `suggested` might hit the balance of "this is likely what you want" to cut down the cognitive load of the developer trying to make a decision in the moment. Would that work?

Reply:
I give up. 20fd594ce7710d0cacf62af4ebf208f2abe2ec5d